Assets/Scripts/Utilities/WadeUtils.cs

```diff
@@ -402,6 +404,36 @@ public static GameObject TempInstantiate(GameObject obj, Vector3 pos, Quaternion
 		return go;
 	}
 
+	public static T GetCopyOf<T>(this Component comp, T other) where T : Component
```

Comment:
Be careful about using reflection in utility code, especially since this is in a class with a bunch of other utility functionality. Reflection isn't supported on all platforms (mostly mobile has problems with it), so when you start using it you risk causing problems when building for release. Probably not important for this game, just something to be aware of.